Assembly code search window in Proxim

alexhrv opened the pull request:
![opfind](https://cloud.githubusercontent.com/assets/2932113/23903297/b997a9e2-08cc-11e7-8829-7f98730812dd.png)

linehill commented:
Seems to work fine.

Couple things needs fixing. The "find operation" command is misleading since it searches more than just an operation, i.e search pattern "ALU.out1 ->" does work. Rename it to "find in disassembly" or just "find". One compiler warning needs fixing:

`FindOperationWindow.hh:79:267: warning: extra ‘;’ [-Wpedantic]`

Otherwise, I approve the merge request. After these fixes, I'll merge if no comments from others.

Some observations made from the feature:
- The next and previous buttons do scroll disassembly window to the matched line vertically but not horizontally.
- Invisible characters can be matched. For example search pattern "-> ALU.in2, ALU.out1" can be found even if the ',' character is not shown in the disassembly window. Same for ';' and possibly others.
- No case sensitivity option. That could be useful.
- Let's say a pattern "add" is searched and matches are found. Now, clearing the search input still displays "x of y matches".

Enhancement idea: let search ignore spaces in the disassembly. For example, allow "RF.1->ALU.in2" fond lines with spaces around "->" string.

alexhrv commented:
Thanks for the comments

> Couple things needs fixing. The "find operation" command is misleading since it searches more than just an operation, i.e search pattern "ALU.out1 ->" does work. Rename it to "find in disassembly" or just "find". One compiler warning needs fixing:

Fixed

> `FindOperationWindow.hh:79:267: warning: extra ‘;’ [-Wpedantic]`

Fixed

> - The next and previous buttons do scroll disassembly window to the matched line vertically but not horizontally.

In wxwidgets I can only scroll to some cell using WxGrid::MakeCellVisible()

> - Invisible characters can be matched. For example search pattern "-> ALU.in2, ALU.out1" can be found even if the ',' character is not shown in the disassembly window. Same for ';' and possibly others.

Fixed

> - No case sensitivity option. That could be useful.

Added this feature

> - Let's say a pattern "add" is searched and matches are found. Now, clearing the search input still displays "x of y matches".

Fixed

> Enhancement idea: let search ignore spaces in the disassembly. For example, allow "RF.1->ALU.in2" find lines with the pattern but spaces around "->" string.

Added this feature
